File: baekjoon/BinarySearch/20551_Sort마스터 배지훈의 후계자.js
const fs = require('fs')
const [nm, ...input] = fs
  .readFileSync('../input.txt')
  .toString()
  .trim()
  .split('\n')
const [n, m] = nm.split(' ').map(Number)

nArr = input.slice(0, n).map((x) => Number(x.trim()))
nArr.sort((a, b) => a - b)
mArr = input.slice(n).map((x) => Number(x.trim()))

function solution() {
  const answer = []
  for (const find of mArr) {
    let start = 0
    let end = n - 1
    let index = -1
    while (start <= end) {
      let mid = parseInt((start + end) / 2)
      if (nArr[mid] === find) {
        index = mid
        end = mid - 1
      } else if (nArr[mid] > find) {
        end = mid - 1
      } else {
        start = mid + 1
      }
    }
    answer.push(index)
  }

  return answer.join('\n')
}

console.log(solution())

File: baekjoon/BinarySearch/20551_Sort마스터 배지훈의 후계자.py
import sys
input = sys.stdin.readline

n, m = list(map(int, input().split()))
nArr = []
mArr = []
for _ in range(n):
    nArr.append(int(input()))
nArr.sort()
for _ in range(m):
    mArr.append(int(input()))


def solution():
    for find in mArr:
        start = 0
        end = len(nArr) - 1
        answer = -1
        while start <= end:
            mid = (start + end) // 2
            if(nArr[mid] == find):
                answer = mid
                end = mid - 1
            elif(nArr[mid] > find):
                end = mid - 1
            else:
                start = mid + 1
        print(answer)


solution()
